### PR TITLE
Tetrate workflow v1

### PR DESCRIPTION
**Pull request by psbrar99**
- Update tetrate-workflow branch with latest scripts so that build happen automatically.
- Remove setup_boring.sh as it is not needed for all supported releases.

### CODE REVIEW

File: .github/workflows/e2e_tests.yml [hunk 1]
```diff
@@ -7,20 +7,17 @@ on:
...
   HUB: ${{ secrets.TEST_HUB }}
-  SHA: "33c75fe4994dfbe01aa01dbefe38e74fdef6025f"
 
 jobs:
   # copied from make_release.yml
   build_fips_proxy:
     # Only run this job when the release type equals "fips".
-    if: contains(github.event.ref, 'fips')
-
+    if: contains(github.event.ref, 'dips')
```
Comment (zirain) [inline .github/workflows/e2e_tests.yml]: dips？nit？

File: .github/workflows/e2e_tests.yml [hunk 2]
```diff
@@ -104,10 +104,166 @@ jobs:
           gsutil cp "${UPSTREAM_GCSP_PATH}/metadata_exchange-${SHA}*" ${RELEASE_GCS_PATH}/
           gsutil cp "${UPSTREAM_GCSP_PATH}/stats-${SHA}*" ${RELEASE_GCS_PATH}/
           gsutil cp "${UPSTREAM_GCSP_PATH}/attributegen-${SHA}*" ${RELEASE_GCS_PATH}/
+          
+  build_fips_arm_proxy:
+    # Do not run this job when the release type equals "fips"as we do not support fips compliant ARM build yet.
+    if: contains(github.event.ref, 'dips')
```
Comment (zirain) [inline .github/workflows/e2e_tests.yml]: ditto

File: .github/workflows/make_release.yml [hunk 3]
```diff
@@ -10,10 +10,10 @@ on:
 jobs:
   build_fips_proxy:
     # Only run this job when the release type equals "fips".
```
Comment (zirain) [inline .github/workflows/make_release.yml:12]: remve this commet?

File: .github/workflows/make_release.yml [hunk 4]
```diff
@@ -97,11 +96,161 @@ jobs:
...
+          cat scripts/release-binary.sh
+
+      - name: Build and push
+        if: ${{ steps.check_already_built.outputs.should_build == '1' }}
+        env:
+          BUILD_WITH_CONTAINER: 1
+          IMG: gcr.io/tetrate-istio-arm/build-tools-proxy:release-${{ steps.get_minor_ver.outputs.RELEASE_VERSION }}-tid
+        run: make push_release
+
+      - name: Put CentOS binary
```
Comment (zirain) [inline .github/workflows/make_release.yml]: I recall upstrem remove support of centos, see https://github.com/istio/proxy/pull/4758

File: .github/workflows/make_release.yml [hunk 5]
```diff
@@ -97,11 +96,161 @@ jobs:
...
+          gsutil cp ${RELEASE_GCS_PATH}/envoy-alpha-${SHA}.tar.gz ${RELEASE_GCS_PATH}/envoy-centos-alpha-${SHA}.tar.gz
+
+      - name: Put Wasm binary
+        # Copy the built Wasm binary in the upstream to the bucket - Wasm is nothing to do with FIPS stuff.
+        env:
+          UPSTREAM_GCSP_PATH: gs://istio-build/proxy
+        run: |
+          SHA=$(git rev-parse --verify HEAD)
+          gsutil cp "${UPSTREAM_GCSP_PATH}/metadata_exchange-${SHA}*" ${RELEASE_GCS_PATH}/
+          gsutil cp "${UPSTREAM_GCSP_PATH}/stats-${SHA}*" ${RELEASE_GCS_PATH}/
```
Comment (zirain) [inline .github/workflows/make_release.yml]: from 1.18 istio move to use native stats filter

File: tetrateci/fips.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+Verify the Go version used by the binaries.
+```shell
+go version pilot-discovery | cut -f2 -d" "
+go version pilot-agent | cut -f2 -d" "
+go version operator | cut -f2 -d" "
+go version istioctl | cut -f2 -d" "
+go version install-cni | cut -f2 -d" "
+```
+The Go version should include `b` to indicate BoringSSL, `go1.16.9b7` for example.
```
Comment (zirain) [inline tetrateci/fips.md]: should include changes since 1.19

File: tetrateci/patches/eks/eks-ingress.1.18.patch [hunk 7]
```diff
@@ -0,0 +1,293 @@
+diff --git a/pkg/test/framework/components/echo/kube/deployment.go b/pkg/test/framework/components/echo/kube/deployment.go
```
Comment (zirain) [inline tetrateci/patches/eks/eks-ingress.1.18.patch]: didn't see this file for v1.17?

File: tetrateci/version_check.py [hunk 8]
```diff
@@ -10,6 +10,11 @@
     "1.11": {"1.18", "1.19", "1.20", "1.21", "1.22"},
     "1.12": {"1.19", "1.20", "1.21", "1.22"}, # officially supported versions according to https://istio.io/latest/news/releases/1.12.x/announcing-1.12
     "1.13": {"1.20", "1.21", "1.22", "1.23"}, # officially supported versions according to https://istio.io/latest/news/releases/1.13.x/announcing-1.13
+    "1.14": {"1.20","1.21", "1.22", "1.23", "1.24"} , # officially supported versions according to https://istio.io/latest/news/releases/1.14.x/announcing-1.14
+    "1.15": {"1.21","1.22", "1.23", "1.24", "1.25"} , # officially supported versions according to https://istio.io/latest/news/releases/1.15.x/announcing-1.15
+    "1.16": {"1.22", "1.23", "1.24", "1.25"} , # officially supported versions according to https://istio.io/latest/news/releases/1.16.x/announcing-1.16
+    "1.17": {"1.22", "1.23", "1.24", "1.25"} , # officially supported versions according to https://istio.io/latest/news/releases/1.17.x/announcing-1.17
+    "1.18": {"1.24", "1.25", "1.26", "1.75"} , # officially supported versions according to https://istio.io/latest/news/releases/1.18.x/announcing-1.18
```
Comment (zirain) [inline tetrateci/version_check.py]:
```suggestion
    "1.18": {"1.24", "1.25", "1.26", "1.27"} , # officially supported versions according to https://istio.io/latest/news/releases/1.18.x/announcing-1.18
```

File: .github/workflows/make_release.yml (hunk 4 above)
Comment (psbrar99) [reply .github/workflows/make_release.yml]: ok, it's currently in master ?
we can remove it once it make it to the release. thanks for heads-up.
https://github.com/istio/istio/blob/1.18.2/bin/init.sh#L53

Comment (zirain) [reply .github/workflows/make_release.yml]: let me double check with john, maybe still WIP.

File: .github/workflows/e2e_tests.yml (hunk 1 above)
Comment (psbrar99) [reply .github/workflows/e2e_tests.yml]: done, thanks

File: .github/workflows/make_release.yml (hunk 5 above)
Comment (psbrar99) [reply .github/workflows/make_release.yml]: added a condition in workflow.

File: .github/workflows/make_release.yml [hunk 9]
```diff
@@ -64,7 +80,7 @@ jobs:
           cp scripts/release-binary.sh scripts/release-binary.sh.tmp
           echo gcloud auth activate-service-account --key-file="/work/${CREDENTIAL_FILE_NAME}" > scripts/release-binary.sh
           cat scripts/release-binary.sh.tmp >> scripts/release-binary.sh
-          sed -i 's/release release\-symbol debug/release release\-symbol/' scripts/release-binary.sh
+          sed  -i 's/for config in release release-symbol asan debug/for config in release/g' scripts/release-binary.sh
```
Comment (zirain) [inline .github/workflows/make_release.yml]:
```suggestion
          sed -i 's/for config in release release-symbol asan debug/for config in release/g' scripts/release-binary.sh
```

File: .github/workflows/make_release.yml [hunk 10]
```diff
@@ -79,63 +95,78 @@ jobs:
...
+          TAG: ${{ steps.get_tag.outputs.TAG }}
         run: |
-          SHA=$(git rev-parse --verify HEAD)
+          SHA=$(echo $TAG | sed 's/test-//g' | sed 's/[0-9].[0-9][0-9].[0-9]-[A-Za-z]*-//g' | sed 's/-.*//g')
           SHOULD_BUILD=1
-          if gsutil ls ${RELEASE_GCS_PATH} | grep ${SHA} ; then
+          if gsutil ls ${RELEASE_GCS_PATH} | grep "${SHA}[^-]" ; then
             SHOULD_BUILD=0
-          fi
+          fi
```
Comment (zirain) [inline .github/workflows/make_release.yml]:
```suggestion
          fi
```

File: tetrateci/patches/eks/eks_1.15.patch [hunk 11]
```diff
@@ -1,8 +1,21 @@
+From a2e8b635a71ac66273edd6388ba883a0214b9cf8 Mon Sep 17 00:00:00 2001
```
Comment (zirain) [inline tetrateci/patches/eks/eks_1.15.patch:1]: for a long term, we can try to push this into upstream if this's a common solution to run test on eks.

**Review by zirain — APPROVED** [review]
Overall LGTM, with small nits.

File: .github/workflows/make_release.yml (hunk 4 above)
Comment (zirain) [reply .github/workflows/make_release.yml]: support for centos7 will be removed in 1.19.

File: tetrateci/patches/eks/eks_1.15.patch (hunk 11 above)
Comment (psbrar99) [reply tetrateci/patches/eks/eks_1.15.patch:1]: sure.

File: .github/workflows/make_release.yml (hunk 4 above)
Comment (psbrar99) [reply .github/workflows/make_release.yml]: ack, thanks for following up!

**Pull request merged**